### DISCUSSION TITLE
gitHub pages odd behaviors with filename extensions

### DISCUSSION THREAD

**Discussion by sflanders95** [community/community#22612] [Other Feature Feedback, Questions, & Ideas]
<p><strong>Behavior:</strong>  test.md file is returning 404 in subdirectory on github pages.</p>
<p><strong>Behavior:</strong>  test.html shows rendering of test.md.  Actual test.html file is ignored.</p>
<p><strong>Behavior:</strong>  test.txt renders test.txt {{ with variables parsed }} inside of _layouts/default.html file and then is converted to source text in the browser similar to view source.</p>
<p><a href="https://sflanders95.github.io/EmpireMillenniumChatBlockerChromeExt/docs/test.md" title="404 is returned" rel="nofollow noopener">https://sflanders95.github.io/EmpireMillenniumChatBlockerChromeExt/docs/test.md</a>  (404)</p>
<p><a href="https://sflanders95.github.io/EmpireMillenniumChatBlockerChromeExt/docs/test.html" title="test.md file is rendered in jekyll-minimal theme" rel="nofollow noopener">https://sflanders95.github.io/EmpireMillenniumChatBlockerChromeExt/docs/test.html</a> (returns test.md &amp; jekyll template, ignores file: test.html in same directory)</p>
<p><a href="https://sflanders95.github.io/EmpireMillenniumChatBlockerChromeExt/docs/test.txt" title="test.txt file is rendered to html and then the source is returned as plain text." rel="nofollow noopener">https://sflanders95.github.io/EmpireMillenniumChatBlockerChromeExt/docs/test.txt</a> (returns test.txt &amp; jekyll template)</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>This is all expected behavior based on how <a href="https://jekyllrb.com/docs/pages/" rel="nofollow noopener">Jekyll processes your repository</a> during its build process. It’s not that the <code>test.html</code> file is ignored, it is just overwritten during the build process by the conversion of <code>test.md</code> to <code>test.html</code>.</p>
<p>I hope that helps!</p>

**Comment by sflanders95** [comment]
<p>It does help thank you.  Since that post I’ve modified the environment so that I can run   jekyll build  and actually see what is being generated in the ./_site/  directory.  That directory I believe is hidden on the github pages site so I didn’t previously notice generated static files &amp; structure.  I was modifying and pushing w/o building locally.  So your response is correct, it works as designed.</p>
<p>Building locally also helps partially explain what is happening to ./docs/test.txt  -  That text file is getting processed with ./_layouts/default.html resulting in ./_site/docs/test.txt  (retaining the .txt extension).  So when the browser requests /docs/test.txt it is getting back the combined documents as html but served as  content-type: text/plain</p>
